src/common/AdaptiveTree.h

```diff
@@ -186,6 +186,54 @@ class ATTR_DLL_LOCAL AdaptiveTree
 
   Settings m_settings;
 
+  class TreeUpdateThread
+  {
+  public:
+    TreeUpdateThread() {}
+    ~TreeUpdateThread();
+
+    void Initialize(AdaptiveTree* tree);
+
+    // \brief Reset start time (make exit the condition variable m_cvUpdInterval and re-start the timeout)
+    void ResetStartTime() { m_cvUpdInterval.notify_all(); }
+
+    // \brief As "std::mutex" lock, but put in pause the manifest updates (support std::lock_guard).
+    //        If an update is in progress, block the code until the update is finished.
+    void lock() { Pause(); }
+
+    // \brief As "std::mutex" unlock, but resume the manifest updates (support std::lock_guard).
+    void unlock() { Resume(); }
+
+  private:
+    void Worker();
+    void Pause();
+    void Resume();
+
+    std::thread* m_updateThread{nullptr};
```

Comment:
Why is this a pointer?

Reply:
please place this question on the other PR, its not part of this